homeassistant/components/template/config_flow.py

```diff
@@ -96,6 +102,9 @@ def generate_schema(domain: str, flow_type: str) -> dict[vol.Marker, Any]:
             ),
         }
 
+    if flow_type == "config":
+        schema = {vol.Required(CONF_NAME): selector.TextSelector()} | schema
```

Comment:
Move this if statement to the beginning so we can initialize the schema directly with the name

```
        schema = {vol.Required(CONF_NAME): selector.TextSelector()}
```